### PR TITLE
Fix duplicate orders on routes and escalation policies

### PR DESCRIPTION
**Pull request by vadimkerr**
# What this PR does

Fix duplicate `order` values for models `EscalationPolicy` and `ChannelFilter` using the same approach as https://github.com/grafana/oncall/pull/2278.

- Make internal API action `move_to_position` a part of [OrderedModelViewSet](https://github.com/grafana/oncall/pull/2568/files#diff-eb62521ccbcb072d1bd2156adeadae3b5895bce6d0d54432a23db3948b0ada54R11-R34), so all ordered model views use the same logic.
- Make public API serializers for ordered models inherit from [OrderedModelSerializer](https://github.com/grafana/oncall/pull/2568/files#diff-d749f94af5a49adaf5074325cdfad10ddd5a52dbfd78b49582867ebb9c92fae5R6-R38), so ordered model views are consistent with each other in public API.
- Remove `order` from plugin fronted, since it's not being used anywhere. The frontend uses step indices & `move_to_position` action instead.
- Make escalation snapshot use step indices instead of orders, since orders are not guaranteed to be sequential (+fix a minor off-by-one bug)

## Which issue(s) this PR fixes

https://github.com/grafana/oncall-private/issues/1680

## Checklist

- [x] Unit, integration, and e2e (if applicable) tests updated
- [x] Documentation added (or `pr:no public docs` PR label added if not required)
- [x] `CHANGELOG.md` updated (or `pr:no changelog` PR label added if not required)

### CODE REVIEW

File: engine/apps/alerts/migrations/0024_auto_20230718_0953.py
Comment (vadimkerr) [inline engine/apps/alerts/migrations/0024_auto_20230718_0953.py:1]: This is similar to [user notification migration](https://github.com/grafana/oncall/blob/eada4a4355084058fd53d97f3b286cd82cad650b/engine/apps/base/migrations/0004_auto_20230616_1510.py) from the previous PR.

File: engine/apps/alerts/migrations/0025_auto_20230718_1042.py
Comment (vadimkerr) [inline engine/apps/alerts/migrations/0025_auto_20230718_1042.py:1]: This is similar to [user notification migration](https://github.com/grafana/oncall/blob/eada4a4355084058fd53d97f3b286cd82cad650b/engine/apps/base/migrations/0004_auto_20230616_1510.py) from the previous PR.

File: engine/apps/api/serializers/channel_filter.py [hunk 1]
```diff
@@ -27,7 +27,6 @@ class ChannelFilterSerializer(OrderedModelSerializerMixin, EagerLoadingMixin, se
         queryset=TelegramToOrganizationConnector.objects, filter_field="organization", allow_null=True, required=False
     )
     telegram_channel_details = serializers.SerializerMethodField()
-    order = serializers.IntegerField(required=False)
```
Comment (vadimkerr) [inline engine/apps/api/serializers/channel_filter.py:30]: `order` is not used by plugin frontend, removing it from the serializer

File: engine/apps/api/views/channel_filter.py [hunk 2]
```diff
@@ -109,24 +112,10 @@ def perform_update(self, serializer):
     @action(detail=True, methods=["put"])
     def move_to_position(self, request, pk):
         instance = self.get_object()
-        position = get_move_to_position_param(request)
-
         if instance.is_default:
             raise BadRequest(detail="Unable to change position for default filter")
 
-        prev_state = instance.insight_logs_serialized
```
Comment (vadimkerr) [inline engine/apps/api/views/channel_filter.py:117]: Changing `order` is now handled by the parent class [OrderedModelViewSet](https://github.com/grafana/oncall/pull/2568/files#diff-eb62521ccbcb072d1bd2156adeadae3b5895bce6d0d54432a23db3948b0ada54R11-R34)

File: grafana-plugin/src/models/channel_filter.ts [hunk 3]
```diff
@@ -4,7 +4,6 @@ import { TelegramChannel } from 'models/telegram_channel/telegram_channel.types'
 
 export interface ChannelFilter {
   id: string;
-  order: number;
```
Comment (vadimkerr) [inline grafana-plugin/src/models/channel_filter.ts:7]: Plugin frontend doesn't use `order` anywhere (it uses step indices and `move_to_position` action instead)

File: engine/apps/alerts/escalation_snapshot/snapshot_classes/escalation_snapshot.py [hunk 4]
```diff
@@ -88,9 +88,7 @@ def executed_escalation_policy_snapshots(self) -> typing.List["EscalationPolicyS
         """
         if self.last_active_escalation_policy_order is None:
             return []
-        elif self.last_active_escalation_policy_order == 0:
-            return [self.escalation_policies_snapshots[0]]
-        return self.escalation_policies_snapshots[: self.last_active_escalation_policy_order]
+        return self.escalation_policies_snapshots[: self.last_active_escalation_policy_order + 1]
```
Comment (vadimkerr) [inline engine/apps/alerts/escalation_snapshot/snapshot_classes/escalation_snapshot.py:91]: Off-by-one bug making `executed_escalation_policy_snapshots` smaller than it should be (it probably affects the escalation auditor task)

File: engine/apps/base/tests/test_ordered_model.py
Comment (joeyorlando) [inline engine/apps/base/tests/test_ordered_model.py:1]: _nit_ if the `OrderedModel` now lives under `./common`, wdyt about moving this to `./common/tests` (or `./common/ordered_model/tests`, up to you)

**Review by joeyorlando — APPROVED** [review]
nice work! the new classes in `common/ordered_model` deduplicate a lot of logic shared between channel filters, escalation policies, and user notification policies 🙌

File: engine/apps/public_api/serializers/personal_notification_rules.py [hunk 5]
```diff
@@ -31,7 +29,8 @@ class PersonalNotificationRuleSerializer(EagerLoadingMixin, serializers.ModelSer
 
     class Meta:
         model = UserNotificationPolicy
-        fields = ["id", "user_id", "position", "type", "duration", "manual_order", "important"]
+        fields = ["id", "user_id", "type", "duration", "important"]
+        fields += ["position", "manual_order"]  # fields from OrderedModelSerializer
```
[the remark this answers is not in the document]
Comment (vadimkerr) [reply engine/apps/public_api/serializers/personal_notification_rules.py]: Great suggestion, applied in https://github.com/grafana/oncall/pull/2568/commits/2d169a29daa4c818f28392a2c42ef5f24148fe99

**Pull request merged**